### ISSUE TITLE
JQ Fails silently with an invalid JSON document

### ISSUE THREAD

**Issue by lukeplausin** [jqlang/jq#1330]
This issue was supposedly fixed already, I'm creating a new issue as #769 and #792 are closed but the issue still appears present in JQ 1.5 (running on macOS 10.11.6).

`jq '.'` returns with exit code 0 and produces no output when run on this document

```
{
  "key1": {
    "unclosedSubkey1":{
      "type":"string"
  }
}
```

**Comment by wtlangford** [comment]
While jq 1.5 does still exhibit this behavior, it is fixed on `master`, where I get this behavior:
```
% echo '{
  "key1": {
    "unclosedSubkey1":{
      "type":"string"
  }
}' | ./jq .
parse error: Unfinished JSON term at EOF at line 7, column 0
```
and an exit code of `4`.

**Comment by gabrielf** [comment]
Version 1.5.2 fails silently on even simpler JSON so it would definitely be nice to get this fix out in a stable release!

```shell
$ # I would expect these to fail in a non-silent way
$ echo '{' | jq '.'
$ echo '{"}' | jq '.'
$ echo '{"key:0}' | jq '.'
```

**Comment by itchyny** [comment]
This bug was already fixed and released in [1.6](https://github.com/jqlang/jq/releases/tag/jq-1.6) in Nov 2018, so closing this issue.
